src/Functions/keyvaluepair/impl/Configuration.cpp

```diff
@@ -13,16 +14,54 @@ namespace ErrorCodes
 namespace extractKV
 {
 
-Configuration::Configuration(char key_value_delimiter_, char quoting_character_, std::vector<char> pair_delimiters_)
-    : key_value_delimiter(key_value_delimiter_), quoting_character(quoting_character_), pair_delimiters(std::move(pair_delimiters_))
+static Configuration::UnexpectedQuotingCharacterStrategy unexpectedQuotingCharacterStrategyFromString(
+    const std::string & unexpected_quoting_character_strategy)
+{
+    const auto unexpected_quoting_character_strategy_lower = Poco::toLower(unexpected_quoting_character_strategy);
+
+    if (unexpected_quoting_character_strategy_lower == "accept")
```

Comment:
Consider using magic_enum - its fun.
In CH it is 'covered' by EnumReflection.h
magic_enum::enum_cast<Configuration::UnexpectedQuotingCharacterStrategy>(unexpected_quoting_character_strategy, magic_enum::case_insensitive);

Reply:
Oh, I didn't know it was included in ClickHouse. I remember using `magic_enum` heavily in ~2018/2019. That's cool.